bazel_tools/scala.bzl

```diff
@@ -442,30 +482,20 @@ def da_scala_library_suite(name, **kwargs):
 
     [rules_scala_library_suite_docs]: https://github.com/bazelbuild/rules_scala/blob/master/docs/scala_library_suite.md
     """
-    _wrap_rule(scala_library_suite, name, **kwargs)
-    _create_scala_source_jar(name = name, **kwargs)
-    _create_scaladoc_jar(name = name, **kwargs)
-
-    if "tags" in kwargs:
-        for tag in kwargs["tags"]:
+    arguments = {}
+    arguments.update(kwargs)
+    arguments = _set_compile_jvm_flags(arguments)
+    _wrap_rule(scala_library_suite, name, **arguments)
+    _create_scala_source_jar(name = name, **arguments)
+    _create_scaladoc_jar(name = name, **arguments)
+
+    if "tags" in arguments:
+        for tag in arguments["tags"]:
             if tag.startswith("maven_coordinates="):
                 fail("Usage of maven_coordinates in da_scala_library_suite is NOT supported", "tags")
                 break
 
-def da_scala_macro_library(**kwargs):
-    """
-    Define a Scala library that contains macros.
-
-    Applies common Scala options defined in `bazel_tools/scala.bzl`.
-    And forwards to `scala_macro_library` from `rules_scala`.
-    Refer to the [`rules_scala` documentation][rules_scala_docs].
-
-    [rules_scala_docs]: https://github.com/bazelbuild/rules_scala#scala_library
-    """
-    _wrap_rule(scala_macro_library, **kwargs)
-    _create_scala_source_jar(**kwargs)
-
-def da_scala_binary(name, unused_dependency_checker_mode = "error", **kwargs):
+def da_scala_binary(name, initial_heap_size = default_initial_heap_size, max_heap_size = default_max_heap_size, **kwargs):
```

Comment:
Probably keeping the docstring makes sense: it links the documentation for the underlying rule and explains why that would be important.

Reply:
I also deleted the function that went with this docstring. 🙂